api/tests/unit/infrastructure/converter/snapshots-csv-converter_test.js

```diff
@@ -1,88 +1,144 @@
-    createdAt: '2017-10-13 09:00:59',
-    studentCode: 'UNIV123',
-    campaignCode: 'CAMPAIGN123',
-    testsFinished: 2,
-    user: {
-      firstName: 'PrenomUser',
-      lastName: 'NomUser',
-    }
-  },
-  {
-    id: 1,
-    score: null,
-    profile: profile2,
-    createdAt: '2017-10-12 16:55:50',
-    studentCode: 'AAA',
-    campaignCode: 'EEE',
-    testsFinished: 1,
-    user: {
-      firstName: 'PrenomUser',
-      lastName: 'NomUser',
-    }
-  }];
-
-  const expectedTextHeadersCSV = '"Nom";"Prenom";"Numero Etudiant";"Code Campagne";"Date";"Score Pix";"Tests Realises";"Interagir";"Securiser l\'environnement numerique"\n';
+  let organization;
+  let competences;
+  let jsonSnapshots;
+
+  beforeEach(() => {
+    organization = new Organization({ type: 'SUP' });
+
+    competences = [
+      new Competence({ id: 21, index: '2.1', name: 'Interagir' }),
+      new Competence({ id: 41, index: '4.1', name: 'Sécuriser l’environnement numérique' }),
+    ];
+
+    const profile1WithEmptyCompetence = '{"included":[{"type":"areas","attributes":{"name":"4. Protection et sécurite"}},{"type":"competences","attributes":{"name":"Securiser l’environnement numerique","index":"4.1","level": 3}},{"type":"competences","attributes":{"name":"Interagir","index":"2.1","level": 4,"course-id":""}}]}';
+
+    const profile2 = {
+      'included': [{
+        'type': 'areas',
+        'attributes': { 'name': '4. Protection et sécurité' }
+      }, {
+        'type': 'competences',
+        'attributes': { 'name': 'Sécuriser l’environnement numérique', 'index': '4.1', 'level': -1 }
+      }, {
+        'type': 'competences',
+        'attributes': { 'name': 'Interagir', 'index': '2.1', 'level': 2, 'course-id': '' }
+      }]
+    };
+
+    jsonSnapshots = [{
+      id: 2,
+      score: '22',
+      profile: profile1WithEmptyCompetence,
+      createdAt: '2017-10-13 09:00:59',
+      studentCode: 'UNIV123',
+      campaignCode: 'CAMPAIGN123',
+      testsFinished: 2,
+      user: {
+        firstName: 'PrenomUser',
+        lastName: 'NomUser',
+      }
+    },
+    {
+      id: 1,
+      score: null,
+      profile: profile2,
+      createdAt: '2017-10-12 16:55:50',
+      studentCode: 'AAA',
+      campaignCode: 'EEE',
+      testsFinished: 1,
+      user: {
+        firstName: 'PrenomUser',
+        lastName: 'NomUser',
+      }
+    }];
+  });
+
+  const expectedTextHeadersCSV = '\uFEFF"Nom";"Prénom";"Numéro Étudiant";"Code Campagne";"Date";"Score Pix";"Tests Réalisés";"Interagir";"Sécuriser l’environnement numérique"\n';
   const expectedTextCSVFirstUser = '"NomUser";"PrenomUser";"UNIV123";"CAMPAIGN123";13/10/2017;22;="2/2";4;3\n';
   const expectedTextCSVSecondUser = '"NomUser";"PrenomUser";"AAA";"EEE";12/10/2017;;="1/2";2;\n';
 
   describe('#convertJsonToCsv()', () => {
 
-    it('should convert an JSON object to a String Object', () => {
+    it('should return a CSV String with at least the headers line', () => {
```

Comment:
Est-ce qu'on ne pourrait pas le nommer du genre : "should return the CSV string with only the header if no snapshots to convert" ... soit en 1 seul bloc 'it', soit avec un context('when no snapshots to convert') ?